bin/ham-bash-lib.sh

```diff
@@ -393,6 +393,72 @@ toolset_import_list() {
     done
 }
 
+# This will always override the MSVC_IDE_DIR.
+# NOTE: This is quite slow and should be used only once during setup.
+toolset_nt_find_ide_dir() {
+    # We assume that all combinations of visual studio versions and editions
+    # can be located in both the x86 and x64 program files directories...
+    local vs_pf_dirs=("$PROGRAMFILES\\Microsoft Visual Studio" \
+        "$PROGRAMW6432\\Microsoft Visual Studio")
+
+    # Version arrays (float, int)
+    # NOTE: the following arrays must be same length
+    local vs_versions_typed=(10.0 2020)
+    local vs_versions_typed_prefix=(' ' '\')
+    local vs_versions_typed_suffix=('' '\Community')
+    local vs_versions_typed_print=('printf "%.1f",' 'print')
+
+    # Count of the supported versions / years
+    # eg. 2020 all the way to 2025
+    # eg. 10.0 all the way to 15.0
+    # NOTE: As of early 2023 community version is at 2022 and Visual Studio at 14.0
+    local vs_versions_supported_count=6
+
+    # loop through dirs, versions to find the IDE dir
+    for vs_pf_dir in "${vs_pf_dirs[@]}"; do
+        for i in "${!vs_versions_typed[@]}"; do
+            local print_=${vs_versions_typed_print[$i]}
+            local prefix_=${vs_versions_typed_prefix[$i]}
+            local suffix_=${vs_versions_typed_suffix[$i]}
+            local version_=${vs_versions_typed[$i]}
+            for y in $(seq 1 $vs_versions_supported_count); do
+                local out_version_=$(awk "BEGIN { $print_ $version_ + ${y} - 1 ; exit }")
```

Comment:
I think `expr` should be a lot faster to do math operation, spawning a subprocess on Windows in horribly slow. For example see https://www.shell-tips.com/bash/math-arithmetic-calculation/#gsc.tab=0